nightfall-deployer/contracts/Config.sol

```diff
@@ -11,12 +11,43 @@ contract Config is Ownable {
     uint256 constant COOLING_OFF_PERIOD = 1 weeks;
     bytes32 constant ZERO = bytes32(0);
 
+    address bootProposer;
+    address bootChallenger;
     mapping(address => uint256) erc20limit;
 
     function initialize() public virtual override initializer {
         Ownable.initialize();
     }
 
+    // restricting proposers
+    modifier onlyBootProposer() {
+        require(msg.sender == bootProposer, 'You are not the boot proposer');
+        _;
+    }
+
+    function setBootProposer(address proposer) external onlyOwner {
```

Comment:
With this approach we can only have one challenger and one proposer address, unless I've misunderstood.  Whilst, initially we want to restrict who can propose and who can challenge, I can image that quite early on we would want to add additional proposers and challengers, if only for resilience (e.g. to guard against a proposer running out of money or getting banned). We can of course upgrade the contract to allow for such a change but would it be good to start with a mapping of allowed proposers and challengers, so that the owner can add and remove them?

Reply:
Well I agree with you, but I think the next step will be to just remove all this, as we're definitely only gonna have one proposer and challenger. But I'll tag @ChaitanyaKonda to decide if it's worth to make it a mapping or if we can leave it as just one proposer and challenger